opal/corelib/runtime.js

```diff
@@ -2831,17 +2886,22 @@
   }
   Opal.return_ivar = function(ivar) {
     return function() {
-      if (this[ivar] == null) this[ivar] = nil;
+      if (this[ivar] == null) {
+        if (this.$$frozen) { return nil; }
+        this[ivar] = nil;
+      }
       return this[ivar];
     }
   }
   Opal.assign_ivar = function(ivar) {
     return function(val) {
+      Opal.deny_frozen_access(this);
       return this[ivar] = val;
     }
   }
   Opal.assign_ivar_val = function(ivar, static_val) {
     return function() {
+      Opal.deny_frozen_access(this);
       return this[ivar] = static_val;
     }
```

Comment:
Some spec was complaining, where it was not caught, but maybe i should check, why it was caught

Reply:
I remember. If the ivar has not been set yet on the object, setting it on the sealed object would throw TypeError, but spec demands FrozenError, thats why. However, i am still researching here, to avoid the dfa.